src/lib.rs

```diff
@@ -122,60 +126,161 @@ pub fn print_render_graph(app: &mut App) {
 /// TODO: Consider adding a means of selecting a schedule other than `Update`.
 pub struct CommandLineArgs;
 
+/// An error from `find_schedule`.
+enum FindScheduleError {
+    /// There was no match. Holds the requested schedule, and the list of valid
+    /// schedules by string.
+    NoMatch(String, Vec<String>),
+    MoreThanOneMatch(String),
+}
+
+impl Debug for FindScheduleError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::NoMatch(request, schedules) => {
+                f.write_fmt(format_args!("No schedules matched the requested schedule '{request}'. The valid schedules are:\n"))?;
+                for schedule in schedules {
+                    f.write_fmt(format_args!("\n{schedule}"))?;
+                }
+                Ok(())
+            }
+            Self::MoreThanOneMatch(request) => f.write_fmt(format_args!(
+                "More than one schedule matched requested schedule '{request}'"
+            )),
+        }
+    }
+}
+
+impl Display for FindScheduleError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        <Self as Debug>::fmt(self, f)
+    }
+}
+
+impl Error for FindScheduleError {}
+
+/// Looks up a schedule by its string name in `App`.
+fn find_schedule<'a>(
+    app: &'a App,
+    schedule_name: &str,
+) -> Result<Interned<dyn ScheduleLabel>, FindScheduleError> {
+    let schedules = app.world().resource::<Schedules>();
+    let schedules = schedules
+        .iter()
+        // Note we get the Interned label from `schedule` since `&dyn ScheduleLabel` doesn't `impl
+        // ScheduleLabel`.
+        .map(|(label, schedule)| (format!("{label:?}"), schedule.label()))
+        .collect::<Vec<_>>();
+
+    let mut found_label = None;
+    for (str, label) in schedules.iter() {
+        if str == schedule_name {
```

Comment:
I'm personally not a fan of making this case insensitive, but if you're really adamant about that, I can do it. Running the command as `cargo r -- dump-schedule update update.dot` looks a little strange to me. I'd rather us be more explicit than less.

We can't do the contains thing since PostUpdate covers Update. You'd never be able to get an unambiguous match for Update.

Reply:
I'll just make this change then, you've convinced me enough to bring me down from feeling strongly to feeling weakly ;)